src/sonic-yang-models/yang-models/sonic-mirror-session.yang

```diff
@@ -0,0 +1,148 @@
+
+    revision 2021-06-15 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-mirror-session {
+
+        container MIRROR_SESSION {
+
+            list MIRROR_SESSION_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
+                            error-message "Invalid mirror session name";
+                            error-app-tag session-name-invalid;
+                        }
+                    }
+                    description
+                        "Mirror session name.";
+                }
+
+                leaf type {
+                    type string {
+                        pattern "ERSPAN|SPAN";
+                    }
+                    default "ERSPAN";
+                    description
+                        "Mirror session type.";
+                }
+
+                leaf src_ip {
+                    type inet:ipv4-address;
+                    description
+                        "ERSPAN source ip. This IP will be set as source ip in
+                        outer header of mirrored frame ";
+                }
+
+                leaf dst_ip {
+                    type inet:ipv4-address;
+                    description
+                        "ERSPAN destination ip. Mirrored frames will be routed to this destination.
+                        This IP will be set as destination ip in outer header of mirrored frame ";
+                }
+
+                leaf gre_type {
+                    type string {
+                        pattern "0[xX][0-9a-fA-F]*";
+                        length 1..6 {
+                            error-message "Invalid GRE type";
+                            error-app-tag gre-type-invalid;
+                        }
+                    }
+                    description
+                        "ERSPAN outer header GRE type.";
+                }
+
+                leaf dscp {
+                    type uint8 {
+                        range "0..63" {
+                            error-message "Invalid dscp value";
+                            error-app-tag dscp-invalid;
+                        }
+                    }
+                    description
+                        "ERSPAN outer header DSCP value. Mirrored frames will be sent with configured DSCP value";
+                }
+
+                leaf ttl {
+                    type uint8 {
+                        range "0..63" {
+                            error-message "Invalid TTL value";
+                            error-app-tag ttl-invalid;
+                        }
+                    }
+                    description
+                        "ERSPAN outer header TTL value. Mirrored frames will be sent with configured TTL value";
+                }
+
+                leaf queue {
+                    type uint8;
```

Comment:
Do we have any valid queue range?

Reply:
This is ASIC specific and we cant have the validation in yang model. 
Please see for more details. https://github.com/Azure/sonic-buildimage/issues/8189